### ISSUE TITLE
The app shutdown behavior should be customizable

### ISSUE THREAD

**Issue by blink1073** [jupyterlab/jupyterlab#8063]
## Description
Currently the app shutdown behavior is provided as a command in the [main menu extension](https://github.com/jupyterlab/jupyterlab/blob/6d86917d74202e04b73459993b30f5a281704910/packages/mainmenu-extension/src/index.ts#L354).  That command cannot easily be overridden by deployments that want to provide a custom shutdown behavior.  For example, they may want to close all open notebooks so that there are no zombie kernels are started the next time the user starts a kernel.

The solution is to pull it out into a separate plugin definition and allow it to be overridden, and document it similar to [copy sharable link](https://github.com/jupyterlab/jupyterlab/blob/master/docs/source/developer/extension_points.rst#copy-shareable-link).

**Comment by jasongrout** [comment]
This sounds like a great idea. We've also had commands hardcoded in the main menu that were not easy to override. Perhaps this should be the general pattern?

**Comment by blink1073** [comment]
I agree.  Our app has a workaround for now to remove the item directly from the file menu by command id.  I'll save this for 3.0.

**Comment by jasongrout** [comment]
Note that a more invasive hack would be going into the command registry and modifying the command to always be not visible or enabled (maybe you can also directly remove it?). That would disable the command system-wide, rather than just from the menu.

**Comment by blink1073** [comment]
Looking at the command registry apis, it isn't possible for someone other than the command registerer to affect a registered command.

**Comment by jasongrout** [comment]
Not unless you throw off the hood and use internal details *evil grin*. I thought we were talking about hacks...
